packages/foundations/scss/colors/classes/_get-class.scss

```
@@ -38,45 +18,37 @@
 	.db-container-color-#{$variant} {
 		@extend %db-#{$variant}-variables;
 
-		color: var(--db-current-color-on-bg-enabled);
-		background-color: var(--db-current-color-bg-lvl-1-enabled);
+		color: var(--db-adaptive-on-bg-basic-emphasis-100-default);
+		background-color: var(--db-adaptive-bg-basic-level-1-default);
 	}
 }
 
 @mixin init-on-background-colors($color) {
 	[data-on-bg-color="#{$color}"],
 	.db-on-bg-color-#{$color} {
-		@if $color == "default" {
-			color: var(--db-current-color-on-bg-enabled);
-		}
-
- @else if $color == "weak" {
-			color: var(--db-current-color-on-bg-weak-enabled);
-		}
-
- @else if $color == "contrast" {
-			color: var(--db-current-color-contrast-high-enabled);
-		}
-
- @else if $color == "contrast-weak" {
-			color: var(--db-current-color-contrast-low-enabled);
-		}
-
- @else if $color == "border" {
-			color: var(--db-current-color-border);
-		}
-
- @else {
+		/* stylelint-disable at-rule-empty-line-before */
+		@if $color == "emphasis-100" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-100-default);
+		} @else if $color == "emphasis-90" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-90-default);
+		} @else if $color == "emphasis-80" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-80-default);
+		} @else if $color == "emphasis-70" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-70-default);
+		} @else if $color == "emphasis-60" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-60-default);
+		} @else if $color == "emphasis-50" {
+			color: var(--db-adaptive-on-bg-basic-emphasis-50-default);
+		} @else {
```

```suggestion
		$emphasis: emphasis-100, emphasis-90, emphasis-80, emphasis-70,
			emphasis-60, emphasis-50;
		@if list.index($emphasis, $color) != null {
			color: var(--db-adaptive-on-bg-basic-#{$color}-default);
		} @else {
```

@nmerget what do you think of this simplification? At the beginning of this file we would need an additional `@use "sass:list";`

---

fixed